### ISSUE TITLE
Regression in entryclone regexp handling

### ISSUE THREAD

**Issue by moewew** [plk/biber#258]
Consider
```latex
\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage[style=numeric,defernumbers,sortcites]{biblatex}
\DeclareBibliographyCategory{related}
\DeclareBibliographyCategory{main}

\DeclareCiteCommand{\cite}[\mkbibbrackets]
  {\usebibmacro{prenote}}
  {\addtocategory{main}{\thefield{entrykey}}%
   \usebibmacro{citeindex}%
   \usebibmacro{cite}}
  {\multicitedelim}
  {\usebibmacro{postnote}}

\DeclareCiteCommand{\citer}[\mkbibparens]
  {\usebibmacro{prenote}}
  {\addtocategory{related}{\thefield{entrykey}}%
   \usebibmacro{citeindex}%
   \usebibmacro{cite}}
  {\multicitedelim}
  {\usebibmacro{postnote}}

\addbibresource{biblatex-examples.bib}

\DeclareSourcemap{
  \maps[datatype=bibtex]{
    \map{
      \perdatasource{biblatex-examples.bib}
      \step[fieldsource=entrykey, match=\regexp{(.*)}, final]
      \step[entryclone={rel-$1}]
    }
  }
}

\begin{document}\nocite{*}
\cite{shore,sarfraz,spiegelberg}

\citer{rel-spiegelberg,rel-springer}

\printbibliography[category=main,title=Main List,resetnumbers]

\DeclareFieldFormat{labelnumberwidth}{\mkbibparens{#1}}
\printbibliography[category=related,resetnumbers,title=Related Documents]

\end{document}
```
from https://github.com/plk/biblatex/issues/297. With `biblatex` 3.12/Biber 2.12 the document gives the expected output.

But when run with the current dev versions from Sourceforge the `rel-` entries are not produced. The `.bcf`s look similar enough, but the `.bbl` from the dev versions does not contain the cloned entries.

The simpler example by David Purton from https://tex.stackexchange.com/q/481491/35864
```latex
\documentclass{article}
\usepackage[defernumbers=true,backend=biber,style=chem-acs,articletitle=true,doi=true]{biblatex}
\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@article{citea,
author = {McClane, John},
journaltitle = {20th Century Fox},
pages = {1--10},
title = {{Die Hard}},
volume = {1},
date = {1988}
}
\end{filecontents}
\addbibresource{\jobname.bib}
\DeclareSourcemap{
  \maps[datatype=bibtex, overwrite]{
    \map{
      \step[fieldsource=entrykey, match=citea, final]
      \step[entryclone={citeb}]
      \step[fieldset=options, fieldvalue={dataonly}]
    }
  }
}
\begin{document}
\nocite{citea}
Yippee Ki Yay, motherf****r.\cite{citeb}
\printbibliography
\end{document}
```
works fine.

**Comment by plk** [comment]
The issue here is the `\perdatasource{biblatex-examples.bib}`. Since the `\addbibresource{biblatex-examples.bib}` is not found as an absolute or relative path, it's resolved by `kpsewhich`. This results in an absolute path but then this is compared to the string `biblatex-examples.bib` which fails. This could be considered a bug but did it work with the previous version or is it perhaps an anomaly of `biblatex-examples.bib` being found by path search in previous attempts?

**Comment by moewew** [comment]
Oh I never thought it would be that...

Indeed, in my tests 
```latex
\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage[style=numeric]{biblatex}


\addbibresource{biblatex-examples.bib}

\DeclareSourcemap{
  \maps[datatype=bibtex]{
    \map{
      \perdatasource{biblatex-examples.bib}
      \step[fieldset=note, fieldvalue={Hello}]
    }
  }
}

\begin{document}
\cite{sigfridsson}

\printbibliography
\end{document}
```
works with Biber 2.12 but not with 3.12 with an otherwise similar set-up.

I'm in two minds about this. Obviously it would be more convenient for me when I'm answering TeX.SX questions if `kpsewhich` files would still match `\perdatasource`. And since it worked before we risk breaking people's documents. On the other hand, I'm not sure how important this is and how complex it would be to fix this on the Biber side.

**Comment by plk** [comment]
I suppose there is a chance that there would be two files with the same name in different places with different content ... this is more of an edge case that the example in question though so this is now fixed in 2.13.

**Comment by moewew** [comment]
Yes, that occurred to me as well. One might have something like `\addbibresource{chapter-one/bib.bib}` and `\addbibresource{chapter-two/bib.bib}`. I'll compare with Biber 2.12 tomorrow if I can come up with a useful test.

**Comment by plk** [comment]
Fixed in latest 2.13 release.
